source/common/common/block_memory_hash_set.h

```diff
@@ -5,7 +5,6 @@
 #include <utility>
 
 #include "envoy/common/exception.h"
-#include "envoy/stats/stats_options.h"
```

Comment:
I think the entirety of block_memory_hash_set can be nuked. We can always restore this later if we need it. I don't think it's used anymore. cc @jmarantz @ambuc in case there any plans to use this in the future.

Reply:
Oh nice! However, I think that's getting into what @jmarantz wanted to leave as a TODO. I tried removing it and most related stuff before realizing it was going to be that far reaching, so I've saved that in a patch in case anyone wants to use it.

Reply:
I think outright file deletions are fine. Let's just go ahead and do it now? Some of these changes really don't make any sense and the code is dead.

Reply:
Done.